src/yetibot/api/google.clj

```diff
@@ -0,0 +1,83 @@
+(ns yetibot.api.google
+  (:require
+   [taoensso.timbre :refer [warn info]]
+   [clojure.string :as s]
+   [clojure.data.json :as json]
+   [clj-http.client :as client]
+   [yetibot.core.config :refer [conf-valid? config-for-ns]]))
+
+(def config (config-for-ns))
+
+(defn configured? []
+  (let [config-keys   [:api-key :custom-search-engine-id]
+        config-values (select-keys config config-keys)]
+    (and (conf-valid? config)
+       (= (count config-values) 2))))
+
+(defonce api-url "https://www.googleapis.com/customsearch/v1?parameters")
+
+;; this has yet to be used somewhere
+(defonce keywords (atom {}))
```

Comment:
- Oh yeah I thought of having the user set stuff like `!google lang en` or `!google domain google.fr` have them stored in keywords 
- and once the user does a `!google search gintama` there would be call that would add it (the atom map) to  query-params before sending it out
-  There would  would something like `!google reset options` to reset preferences
- The `accepted-keywords` was just to make the code easier to do rather than have `n`
functions do stuff 

I think this is a bit long  and winded. Not sure what I think of doing stuff as you did with jira (?) where you just add flags  example -> `!google search --lang en -w something mars bars`